### DISCUSSION TITLE
update from 2.3.2 to 2.3.21 fails

### DISCUSSION THREAD

**Discussion by veronikasen** [Security-Onion-Solutions/securityonion#2532] [Unsupported Versions]
I'm trying to to 2.3.21, I downloaded the new ISO and did an airgap install,
 but I get the following error:

 ***  WARN: Detected -r or -R option while installing Salt packages for Python 3.
 *  WARN: Python 3 packages for older Salt releases requires the EPEL repository to be installed.
 *  WARN: Installing the EPEL repository automatically is disabled when using the -r or -R options.
Loaded plugins: fastestmirror, versionlock
Determining fastest mirrors
https://SecurityOnion-Master/repo/repodata/repomd.xml: [Errno 14] curl#22 - "The requested URL returned error: 502"
Trying other mirror.

 One of the configured repositories failed (Airgap Repo),
 and yum doesn't have enough cached data to continue. At this point the only
 safe thing yum can do is fail. There are a few ways to work "fix" this:

     1. Contact the upstream for the repository and get them to fix the problem.

     2. Reconfigure the baseurl/etc. for the repository, to point to a working
        upstream. This is most often useful if you are using a newer
        distribution release than is supported by the repository (and the
        packages for the previous distribution release still work).

     3. Run the command with the repository temporarily disabled
            yum --disablerepo=airgap_repo ...

     4. Disable the repository permanently, so yum won't use it by default. Yum
        will then just ignore the repository until you permanently enable it
        again or use --enablerepo for temporary usage:

            yum-config-manager --disable airgap_repo
        or
            subscription-manager repos --disable=airgap_repo

     5. Configure the failing repository to be skipped, if it is unavailable.
        Note that yum will try to contact the repo. when it runs most commands,
        so will have to try and fail each time (and thus. yum will be be much
        slower). If it is a very temporary problem though, this is often a nice
        compromise:

            yum-config-manager --save --setopt=airgap_repo.skip_if_unavailable=true

failure: repodata/repomd.xml from airgap_repo: [Errno 256] No more mirrors to try.
https://SecurityOnion-Master/repo/repodata/repomd.xml: [Errno 14] curl#22 - "The requested URL returned error: 502"
 * ERROR: Failed to run install_centos_stable_deps()!!!
Applying yum versionlock for Salt.

Loaded plugins: fastestmirror, versionlock
Adding versionlock on: 0:salt-minion-3001.1-1.el7
Adding versionlock on: 0:salt-master-3001.1-1.el7
Adding versionlock on: 0:salt-3001.1-1.el7
versionlock added: 3
Checking if Salt was upgraded.

Salt upgrade failed. Check of indicators of failure in /root/soup.log.
Once the issue is resolved, run soup again.
Exiting.**

when i disabled the airgap repo, i got another error - what should I do?

**Comment by TOoSmOotH** [comment]
Can you check the dates on the file in /nsm/repo? Did they get copied over when you ran soup?

**Reply by veronikasen** [reply]
The dates of the files are old, before I even Installed SO at first

**Comment by ernstdieter** [comment]
Hi, I had a similar issue. I added in my proxy setting under /etc/environment:
no_proxy="IP_of_SO_Master,Name_of_SO_Master"
reload the proxy settings and it was working.

BR
